tests/sendAudit.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+var ATNA = require('../index');
+var tap = require('tap');
+var fs = require('fs');
+var dgram = require('dgram');
+var net = require('net')
+var tls = require('tls')
+
+var setupConfig = function (callback) {
+  var connDetails = {
+    interface: 'udp',
+    host: 'localhost',
+    port: 5050,
+    options: {
+      key: fs.readFileSync('./tests/resources/client-tls/key.pem').toString(),
+      cert: fs.readFileSync('./tests/resources/client-tls/cert.pem').toString(),
+      ca: fs.readFileSync('./tests/resources/server-tls/cert.pem').toString(),
+    }
+  }
+
+  callback(connDetails)
+}
+
+var setupUDPserver = function (port, callback) {
+  var server = dgram.createSocket('udp4');
+
+  server.bind({
+    port: port
+  });
+
+  server.on('listening', function() {
+    callback()
+  });
+
+  server.on('message', function(msg, rinfo) {
+    server.close();
+  })
+
+  server.on('error', function (err) {
+    callback(err)
+  });
+}
+
+var setupTLSserver = function (port, callback) {
+  var options = {
+    key: fs.readFileSync('./tests/resources/server-tls/key.pem').toString(),
+    cert: fs.readFileSync('./tests/resources/server-tls/cert.pem').toString(),
+    requestCert: true,
+    rejectUnauthorized: false,
+    secureProtocol: 'TLSv1_method'
+  };
+
+  var server = tls.createServer(options, function(sock) {
+    sock.on('data', function(data) {
+      server.close();
+    });
+  });
+
+  server.listen(port, 'localhost', function() {
+    callback();
+  });
+}
+
+var setupTCPserver = function (port, callback) {
+  var server = net.createServer(function(sock) {
+    sock.on('data', function(data) {
+      server.close();
+    });
+  });
+
+  server.listen(port, 'localhost', function() {
+    callback();
+  });
+}
+
+tap.test('should throw an error for an invalid interface type', function (t) {
+  setupConfig(function (config) {
+    config.interface = 'NOTVALID'
+
+    ATNA.send.sendAuditEvent('This is a test message', config, function (err) {
```

Comment:
These tests don't really check that the servers got the correct message. It would be useful to make sure of this. You could do this by making the callback only return when data is received (and return that data) or an error occurs.